engine/resource/src/providers/provider_zip.cpp

```diff
@@ -74,52 +74,15 @@ static void DeleteZipArchiveInternal(dmResourceProvider::HArchiveInternal _archi
 
 static void CreateEntryMap(ZipProviderContext* archive)
 {
-    uint32_t count = archive->m_Manifest->m_DDFData->m_Resources.m_Count;
-    archive->m_EntryMap.SetCapacity(dmMath::Max(1U, (count*2)/3), count);
-
-    dmLiveUpdateDDF::HashAlgorithm algorithm = archive->m_Manifest->m_DDFData->m_Header.m_ResourceHashAlgorithm;
-    uint32_t hash_len = dmResource::HashLength(algorithm);
-
-    dmHashTable32<bool> added;
-    added.SetCapacity(dmMath::Max(1U, (count*2)/3), archive->m_EntryMap.Capacity());
-
     dmZip::HZip zip = archive->m_Zip;
-    for (uint32_t i = 0; i < count; ++i)
-    {
-        dmLiveUpdateDDF::ResourceEntry* entry = &archive->m_Manifest->m_DDFData->m_Resources.m_Data[i];
-
-        char hash_buffer[dmResourceArchive::MAX_HASH*2+1];
-        dmResource::BytesToHexString(entry->m_Hash.m_Data.m_Data, hash_len, hash_buffer, sizeof(hash_buffer));
-        hash_buffer[dmResourceArchive::MAX_HASH*2] = 0;
-
-        dmZip::Result zr = dmZip::OpenEntry(archive->m_Zip, hash_buffer);
-        if (dmZip::RESULT_OK != zr)
-        {
-            continue;
-        }
-
-        EntryInfo info;
-        info.m_ManifestEntry = entry;
-        info.m_Size = entry->m_Size;
-        dmZip::GetEntryIndex(zip, &info.m_EntryIndex);
-
-        // For live update resources, we account for the header size
-        dmZip::CloseEntry(archive->m_Zip);
+    uint32_t archive_entry_count = dmZip::GetNumEntries(zip);
 
-        archive->m_EntryMap.Put(entry->m_UrlHash, info);
-        added.Put(info.m_EntryIndex, true);
+    dmHashTable64<EntryInfo> temp_archive_map;
+    temp_archive_map.SetCapacity(dmMath::Max(1U, (archive_entry_count * 2) / 3), archive_entry_count);
 
-        DM_RESOURCE_DBG_LOG(3, "Added entry: %s %llx (%u bytes)\n", hash_buffer, entry->m_UrlHash, info.m_Size);
-    }
-
-    // Also add any other files that the developer might have added to the zip archive
-    uint32_t entry_count = dmZip::GetNumEntries(zip);
-    for (uint32_t i = 0; i < entry_count; ++i)
+    // Itereate archive and collect all the info about files
+    for (uint32_t i = 0; i < archive_entry_count; ++i)
```

Comment:
We do a single archive iteration, reading all the files information and collect data in a temp map.

Reply:
Itereate -> Iterate